sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -119,23 +146,142 @@ static void *_event_loop(void *arg)
     memset(&local, 0, sizeof(sock_udp_ep_t));
     local.family = AF_INET6;
     local.netif  = SOCK_ADDR_ANY_NETIF;
-    local.port   = CONFIG_GCOAP_PORT;
-
+    if (IS_ACTIVE(CONFIG_GCOAP_ENABLE_DTLS)) {
+        local.port = CONFIG_GCOAPS_PORT;
+    } else {
+        local.port = CONFIG_GCOAP_PORT;
+    }
     int res = sock_udp_create(&_sock_udp, &local, NULL, 0);
     if (res < 0) {
         DEBUG("gcoap: cannot create sock: %d\n", res);
         return 0;
     }
 
     event_queue_init(&_queue);
-    sock_udp_event_init(&_sock_udp, &_queue, _on_sock_evt, NULL);
-    event_loop(&_queue);
 
+    if (IS_ACTIVE(CONFIG_GCOAP_ENABLE_DTLS)) {
+#if IS_ACTIVE(CONFIG_GCOAP_ENABLE_DTLS)
+        sock_dtls_init();
```

Comment:
that's already called by `sys/auto_init/auto_init.c`

Reply:
True, but I still get segmentation fault, if I remove this line and try to establish a session with gcoap. Need to debug here :/

Reply:
Fixed, but now depends on https://github.com/RIOT-OS/RIOT/pull/16528